.github/workflows/reusable-pre-commit.yml

```diff
@@ -0,0 +1,28 @@
+name: Pre-commit
+# The pre-commit configuration is in .pre-commit-config.yaml
+# we don't use the pre-commit action because it doesn't support local hooks in its virtual environment
+
+on:
+  workflow_call:
+    inputs:
+      ros_distro:
+        description: 'ROS2 distribution name'
+        required: true
+        type: string
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ros-tooling/setup-ros@0.7.1
+    - uses: actions/cache@v4
+      with:
+        path: ~/.cache/pre-commit
+        key: pre-commit-3|${{ inputs.ros_distro }}|${{ hashFiles('.pre-commit-config.yaml') }}
+    - name: Install system hooks and run pre-commit
+      run: |
+        sudo apt-get install -qq ros-${{ inputs.ros_distro }}-ament-cppcheck ros-${{ inputs.ros_distro }}-ament-cpplint ros-${{ inputs.ros_distro }}-ament-lint-cmake ros-${{ inputs.ros_distro }}-ament-copyright
+        source /opt/ros/${{ inputs.ros_distro }}/setup.bash
+        python -m pip install pre-commit
```

Comment:
This will probably get problematic on 24.04, since python 3.11 doesn't allow to install pip packages without a venv by default (see e.g. https://stackoverflow.com/questions/75602063/pip-install-r-requirements-txt-is-failing-this-environment-is-externally-mana). We can keep that in mind for later, though...

Reply:
For me it seemed that every step has its own vritual environment? this is why https://github.com/pre-commit/action/blob/main/action.yml#L13 didn't show any of the ROS packages before?

Reply:
yes, the hooks use their own venv (if they are python hooks). However, `pre-commit` itself is installed without a venv here. I am simply referring to the `python -m pip install pre-commit` command. This will not work on 24.04 (without any modifications to the systemat least). One way to resolve this (Doing this on my 23.10 machine for instance) would be to install `pipx (available in apt)` and then do `pipx install pre-commit && pipx ensurepath`

Reply:
then we will have more problems, for example also with the sphinx build I guess
https://github.com/ros-controls/control.ros.org/blob/51ff1372c4014845b4b3b7f9fd86644d05fa1db3/.github/workflows/sphinx-make-page.yml#L22-L32

let's keep it like this for now and remember it until rolling is targeting on 24.04 (which might be together with Jazzy release?)